### ISSUE TITLE
numpy raise a Singular matrix error

### ISSUE THREAD

**Issue by kaiwang0112006** [shahejokarian/regression-prediction-interval#1]
How to mathematically handle this error in the code?

**Comment by shahejokarian** [comment]
If I understood the error correct, you have singular matrix and singular matrices are not invertible, that's why most probably the error should have raised by this line np.linalg.inv(...).
Try to read about [Moore Penrose](https://en.wikipedia.org/wiki/Moore%E2%80%93Penrose_pseudoinverse) maybe it might be helpful.
And here is the numpy function [documentation](https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.pinv.html#numpy.linalg.pinv)

**Comment by kaiwang0112006** [comment]
Yes, I mean is it possible to write a try except block to do prediction anyway
